MeshLib/ElementCoordinatesMappingLocal.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#include "ElementCoordinatesMappingLocal.h"
+
+#include <limits>
+#include <cassert>
+
+#include "GeoLib/AnalyticalGeometry.h"
+
+#include "MathLib/MathTools.h"
+#include "MeshLib/Node.h"
+
+namespace MeshLib
+{
+
+ElementCoordinatesMappingLocal::ElementCoordinatesMappingLocal(
+    const Element& e,
+    const CoordinateSystem &global_coords)
+: _coords(global_coords)
+{
+    assert(e.getDimension() <= global_coords.getDimension());
+    for(unsigned i = 0; i < e.getNNodes(); i++)
+        _point_vec.push_back(new MeshLib::Node(*(e.getNode(i))));
+
+    getRotationMatrixToGlobal(e, global_coords, _point_vec, _matR2global);
+    rotateToLocal(_matR2global.transpose(), _point_vec);
+}
+
+ElementCoordinatesMappingLocal::~ElementCoordinatesMappingLocal()
+{
+	for (auto p : _point_vec) delete p;
+}
+
+void ElementCoordinatesMappingLocal::rotateToLocal(
+    const RotationMatrix &matR2local,
+    std::vector<MeshLib::Node*> &vec_pt) const
+{
+    for(unsigned i = 0; i < vec_pt.size(); i++)
+        vec_pt[i]->setCoords((matR2local* (*vec_pt[i])).getCoords());
```

Comment:
More style than performance. If I'm not wrong this could be
```c++
for(MeshLib::Node* node : vec_pt)
    node->setCoords((matR2local* (*node)).getCoords());
```
Also there is a mismatch between vector of Node (pointers) and its name vec_pt.